src/ranking.jl

```diff
@@ -168,7 +171,23 @@ of an array. The `lt` keyword allows providing a custom "less
 than" function; use `rev=true` to reverse the sorting order.
 Items that compare equal receive the mean of the
 rankings they would have been assigned under ordinal ranking.
+Missing values are assigned rank `missing`.
 """
 tiedrank(x::AbstractArray; lt = isless, rev::Bool = false) =
     tiedrank!(Array{Float64}(uninitialized, size(x)), x, sortperm(x; lt = lt, rev = rev))
 
+for (f, f!, S) in zip([:ordinalrank, :competerank, :denserank, :tiedrank],
+                      [:ordinalrank!, :competerank!, :denserank!, :tiedrank!],
+                      [Int64, Int64, Int64, Float64])
+    @eval begin
+        function $f(x::AbstractArray{Union{T, Missing}}; lt = isless, rev::Bool = false) where {T}
+            xv = collect(skipmissing(x))
+            sp = sortperm(xv; lt = lt, rev = rev)
+            rks = missings($S, length(x))
+            $(f!)(view(rks, .!ismissing.(x)), xv, sp)
```

Comment:
Since you need to compute `.!ismissing.(x)`, better do something like `mask = .!ismissing.(x); xv = x[mask]` above, that could be more efficient, and preserve the array type. Or maybe you can use a view there too?

Reply:
In that case don't I get a performance penalty as `x[mask]` has eltype `Union{T, Missing}` ? I thought that was to be avoided as `sortperm` is probably the most costly operation.

Reply:
Maybe `disallowmissing(view(x, mask))` is the best solution.

Reply:
Ah, right, and `disallowmissing` will have to check that all entries are non-missing, so that's not more efficient either. Better keep the current code then.